bdsf/readimage.py

```diff
@@ -329,21 +335,24 @@ def beam2pix(x):
             th = bpa
             return (s1, s2, th)
 
-        def pix2beam(x):
+        def pix2beam(x, is_error=False):
             """ Converts beam in pixels to deg. Use when no dependence on
             position is appropriate.
 
             Input beam angle should be degrees CCW from the +y axis of the image.
             The output beam angle is degrees CCW from North at image center.
+            Set is_error = True when x gives the errors on the parameters instead of
+                the parameters themselves.
             """
             s1, s2, th = x
             bmaj = abs(s1 * cdelt1)
             bmin = abs(s2 * cdelt2)
             bpa = th
-            if bmaj < bmin:
-                bmaj, bmin = bmin, bmaj
-                bpa += 90.0
-            bpa = divmod(bpa, 180)[1] ### bpa lies between 0 and 180
+            if not is_error:
+                if bmaj < bmin:
+                    bmaj, bmin = bmin, bmaj
+                    bpa += 90.0
+                bpa = divmod(bpa, 180)[1] ### bpa lies between 0 and 180
```

Comment:
This code snippet also appeared in `bdsf/readimage.py`. Maybe turn it into a separate utility function?

Reply:
Good point -- I see it three times in fact. I made a function for it (in `functions.py`).